haiku-apps/alivejournal/alivejournal-1.61.recipe

```diff
@@ -0,0 +1,47 @@
+SUMMARY="Client for LiveJournal online journalling community"
+DESCRIPTION="AliveJournal is a revamped version of Simon Huet's original client."
+HOMEPAGE="https://github.com/grahams/alivejournal"
+COPYRIGHT="2000 Simon Huet
+	2010 Sean Graham"
+LICENSE="GNU GPL v2"
+REVISION="1"
+gitSrcRevision="e3d2c76f0645076f4e05ea942427e0f6e919948b"
+SOURCE_URI="https://github.com/grahams/alivejournal/archive/$gitSrcRevision.tar.gz"
+CHECKSUM_SHA256="8215d4ec005fd345c15ece8223f022cc6d434d940dc3c699a00c87c4e8b6ac49"
+SOURCE_FILENAME="alivejournal-$portVersion-$gitSrcRevision.tar.gz"
+SOURCE_DIR="alivejournal-$gitSrcRevision"
+
+ARCHITECTURES="x86_gcc2 x86_64"
+
+PROVIDES="
+	alivejournal = $portVersion
+	app:AliveJournal = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:g++
+	cmd:make
+	cmd:sed
+	"
+
```

Comment:
On x86_64 (and x86) STDCPPLIBS is set to stdc++ but on x86_gcc2 it is set to stdc++.r4 so we should fix the Makefile. This can be done with a PATCH function (before BUILD).
```
PATCH()
{
	sed -i -e '/^LIBS=/ s/stdc++/$(STDCPPLIBS)/' Makefile
}
```
I don't know, however, if it would be OK to upstream the resulting change.

Reply:
To be honest, I feel guilty bothering the upstream author. I think it's best to use this command for now.

Reply:
I also think it is better to use the PATCH function.
The first reason is that I don't know who sets it, makefile_engine or haikuporter. Well I think it is makefile_engine.
The second reason is that libstdc++.r4 will disapear some day, while libstdc++ will remain.
So, since the Makefile on the upstream repo already works fine with x86_64, as soon as x86_gcc2 is phased out, upstream will be OK on all supported architectures.